js/ui/viewSelector.js

```diff
@@ -211,7 +211,9 @@ const ViewsDisplayLayout = new Lang.Class({
 
         let gridContainerBox = allocation.copy();
         gridContainerBox.y1 = this._computeGridContainerPlacement(gridContainerHeight, entryHeight, availHeight);
-        gridContainerBox.y2 = Math.min(gridContainerBox.y1 + gridContainerHeight, allocation.y2);
+        // If the icon grid's current height is 0 we still want to give its box the available height
```

Comment:
This seems a bit confusing to read. Also, it seems wrong: `gridContainerHeight` plays a part in calculating the `y1` coordinate just above, so it looks like ignoring that here and use the allocation.y2 value when the determined height for the grid is zero could cause some trouble. Are you sure this works as expected?